scripts/o3de/o3de/engine_template.py

```diff
@@ -226,50 +259,96 @@ def _execute_restricted_template_json(json_data: dict,
                 restricted_json_data.update({"restricted_name": destination_name})
                 s.write(json.dumps(restricted_json_data, indent=4) + '\n')
 
+    ###################################################################################
+    # for each createDirectories in the template copy any entries in the json_data that are for this platform
+    for create_directory in template_json_data['createDirectories']:
+        new_dir = pathlib.Path(create_directory['dir'])
+        if not keep_restricted_in_instance and 'Platform' in new_dir.parts:
+            try:
+                # the name of the Platform should follow the '/Platform/'
+                pattern = r'/Platform/(?P<Platform>[^/:*?\"<>|\r\n]+/?)'
+                found_platform = re.search(pattern, new_dir.as_posix()).group('Platform')
+            except Exception as e:
+                pass
+            else:
+                found_platform = found_platform.replace('/', '')
+                if found_platform == restricted_platform:
+                    create_dirs = []
+                    if 'createDirectories' in json_data.keys():
+                        create_dirs = json_data['createDirectories']
+                    create_dirs.append(create_directory)
+                    json_data.update({'createDirectories': create_dirs})
+
+    # for each copyFiles in the template copy any entries in the json_data that are for this platform
+    for copy_file in template_json_data['copyFiles']:
+        new_file = pathlib.Path(copy_file['file'])
+        if not keep_restricted_in_instance and 'Platform' in new_file.parts:
+            try:
+                # the name of the Platform should follow the '/Platform/'
+                pattern = r'/Platform/(?P<Platform>[^/:*?\"<>|\r\n]+/?)'
+                found_platform = re.search(pattern, new_file.as_posix()).group('Platform')
```

Comment:
This should use the `extract_platform_name` method suggested in the previous comment